### PR TITLE
Cleanup more unused deserialization logic

### PR DESCRIPTION
**Pull request by original-brownbear**
Same as #107181 just another round of removing now unused deserialization constructors and making fields final where possible as a result.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java [hunk 1]
```diff
@@ -291,32 +288,13 @@ public Builder setDocs(List<Map<String, Object>> docs) {
...
-                return this;
-            }
-
-            public Builder setSkipQueue(boolean skipQueue) {
-                this.skipQueue = skipQueue;
-                return this;
-            }
-
             public Request build() {
-                return new Request(id, update, docs, textInput, skipQueue, timeout);
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java:319]: Given these parameters are still needed in the Request object, I think it makes sense to keep them around for documentation/future use...

**Comment by thecoop** [comment]
Several of these still have valid serialization methods. Can we get rid of the serialization code as well?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java (hunk 1 above)
Comment (original-brownbear) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java:319]: Heh actually turns out the whole builder class is unused :) => and it's gone.

**Comment by original-brownbear** [comment]
> Can we get rid of the serialization code as well?

That is a good question :) One for core/infra I'd say. I bet some of these could (theoretically) still be needed for interaction with the transport client. I have no idea if that's true or if we even still do or want to support the 7.x transport client communicating with an 8.x cluster?
If we don't care about the transport client, not only could we drop the serialization methods, we could actually drop the transport actions associated with these message objects and inline them into the REST actions etc. that make use of them! That would be a massive code saving :)

**Comment by DaveCTurner** [comment]
> If we don't care about the transport client, not only could we drop the serialization methods, we could actually drop the transport actions associated with these message objects and inline them into the REST actions etc. that make use of them!

That's not quite true, the REST/transport layer boundary is still where (most of) the security checks occur. And it's the seam at which we split things apart for a lot of testing too.

**Comment by thecoop** [comment]
If the deserialization code isn't used, are there situations the serialization code might be used by itself? If the object is never deserialized, then it is never serialized.

**Comment by DaveCTurner** [comment]
However to answer Simon's question, IMO yes we can drop any unused serialization code here too. The transport client is not at all supported against 8.x.

**Comment by original-brownbear** [comment]
> The transport client is not at all supported against 8.x.

:O didn't realise that, lets do that in a follow up, that's a massive cleanup incoming :)

> That's not quite true, the REST/transport layer boundary is still where (most of) the security checks occur. And it's the seam at which we split things apart for a lot of testing too.

Right right but theoretically speaking, we could get rid of that artificial boundary. Also, in some cases I believe we could do so practically because the transport action just fans out to other stuff that's then authenticated individually yet again. But certainly something to handle with care I admit it :)

**Comment by DaveCTurner** [comment]
> If the deserialization code isn't used, are there situations the serialization code might be used by itself? If the object is never deserialized, then it is never serialized.

That argument isn't totally watertight, there could be situations where we're required to send a particular kind of message (e.g. a response to an older remote node) but will never receive such a message (e.g. because we never make the corresponding request).

**Comment by DaveCTurner** [comment]
> Also, in some cases I believe we could do so practically because the transport action just fans out to other stuff that's then authenticated individually yet again.

True, but IMO it'd be better to do the authz once at the outer layer and then skip it on the fanned-out actions (going from N+1 authz events to 1) rather than skipping it at the outer layer (which goes from N+1 to N, not a huge deal)

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java [hunk 2]
```diff
@@ -111,15 +109,14 @@ public static Request forTextInput(String id, InferenceConfigUpdate update, List
...
-            boolean highPriority,
             TimeValue inferenceTimeout
         ) {
             this.id = ExceptionsHelper.requireNonNull(id, InferModelAction.Request.DEPLOYMENT_ID);
             this.docs = docs;
             this.textInput = textInput;
             this.update = update;
             this.inferenceTimeout = inferenceTimeout;
-            this.highPriority = highPriority;
+            this.highPriority = false;
```
Comment (thecoop) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java:119]: ...it's already false?

**Review by thecoop — APPROVED** [review]
One nit, but otherwise good. Nuke'em

**Pull request merged**